File: examples/proto/csharp/TaskExample.cs
using System;
using Monarchic.AgentProtocol.V1;

public class TaskExample {
  public static void Main(string[] args) {
    var task = new Task {
      Version = "v1",
      TaskId = "task-123",
      Role = AgentRole.Dev,
      Goal = "Implement protocol"
    };

    Console.WriteLine(task);
  }
}

File: examples/proto/objective-c/TaskExample.m
#import <Foundation/Foundation.h>
#import "MonarchicAgentProtocol.pbobjc.h"

int main(int argc, const char * argv[]) {
  @autoreleasepool {
    MAPTask *task = [MAPTask message];
    task.version = @"v1";
    task.taskId = @"task-123";
    task.role = MAPAgentRoleDev;
    task.goal = @"Implement protocol";

    NSLog(@"%@", task);
  }
  return 0;
}
